api/server/router/image/image_routes.go

```diff
@@ -67,7 +71,61 @@ func (ir *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrit
 		progressErr = ir.backend.PullImage(ctx, img, tag, platform, metaHeaders, authConfig, output)
 	} else { // import
 		src := r.Form.Get("fromSrc")
-		progressErr = ir.backend.ImportImage(ctx, src, repo, platform, tag, message, r.Body, output, r.Form["changes"])
+
+		var ref reference.Named
+		if repo != "" {
+			var err error
+			ref, err = reference.ParseNormalizedNamed(repo)
+			if err != nil {
+				return errdefs.InvalidParameter(err)
+			}
+			if _, isCanonical := ref.(reference.Canonical); isCanonical {
```

Comment:
Not new (existing code did this as well), but wondering if this should look for `.Digested`; I recall I had a PR open for that in another area; https://github.com/moby/moby/pull/44510

Reply:
Yeah, good point, I recently did consider this in other context, but didn't remember that this one also does this. Fixed, thanks!